src/plugins/intel_cpu/src/nodes/rope.cpp

```diff
@@ -7,17 +7,281 @@
 #include "common/bfloat16.hpp"
 #include "common/cpu_memcpy.h"
 #include "cpu/x64/cpu_isa_traits.hpp"
+#include "cpu/x64/jit_generator.hpp"
+#include "cpu/x64/jit_uni_eltwise.hpp"
+#include "dnnl_extension_utils.h"
+#include "emitters/plugin/x64/jit_load_store_emitters.hpp"
+#include "openvino/core/type/element_type.hpp"
 #include "shape_inference/shape_inference_internal_dyn.hpp"
 #include "utils/plain_tensor.hpp"
 
 #include <chrono>
 #include <string>
 #include <vector>
 
+using namespace dnnl;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::x64;
+using namespace dnnl::impl::utils;
+using namespace Xbyak;
+
 namespace ov {
 namespace intel_cpu {
 namespace node {
 
+struct jit_rotary_compile_params {
+    ov::element::Type src_prc;
+    ov::element::Type dst_prc;
+    size_t rotary_ndims;
+    bool interleave;
+    bool mix_cos_sin;
+};
+struct jit_rotary_call_args {
```

Comment:
minor: please add const qualifier to the input pointer

Reply:
Done.